aiida/tools/importexport/common/archive.py

```diff
@@ -40,59 +60,135 @@ class Archive(object):
             archive.version
 
     """
+    MANDATORY_DATA_KEYS = {'node_attributes', 'node_extras', 'export_data', 'links_uuid', 'groups_uuid'}
+    MANDATORY_METADATA_KEYS = {'aiida_version', 'export_version', 'all_fields_info', 'unique_identifiers'}
 
-    FILENAME_DATA = 'data.json'
-    FILENAME_METADATA = 'metadata.json'
-
-    def __init__(self, filepath):
+    def __init__(self, filepath, output_filepath=None, overwrite=False, silent=True, sandbox_in_repo=True):
         self._filepath = filepath
+        self._output_filepath = output_filepath
+        self._overwrite = overwrite
+        self._silent = silent
+        self._sandbox_in_repo = sandbox_in_repo
         self._folder = None
         self._unpacked = False
+        self._keep = False
         self._data = None
         self._meta_data = None
+        self._archive_format = None
```

Comment:
You are not using this consistently. For starters, I cannot find where you _set_ this. But then you _do_ use the value to switch on in `unpack` and `repack`. You are also sometimes using the attribute and sometimes the property

Reply:
I set it in `unpack`, where it is set to `zip`, unless the archive was a `tar` file, in which case it is set to `tar`.

I will go through its uses. Also, I think this is an attribute that should be completely reworked from a design point of view, in order to make it possible to choose the archive you are outputting (if repacking).